tasks/linter.py

```diff
@@ -369,25 +370,57 @@ def is_get_parameter_call(file):
 def gitlab_ci(ctx, test="all", custom_context=None):
     """
     Lint Gitlab CI files in the datadog-agent repository.
+
+    This will lint the main gitlab ci file with different
+    variable contexts and lint other triggered gitlab ci configs.
     """
-    all_contexts = []
-    if custom_context:
-        all_contexts = load_context(custom_context)
-    else:
-        all_contexts = get_preset_contexts(test)
-    print(f"We will tests {len(all_contexts)} contexts.")
+
     agent = get_gitlab_repo()
-    for context in all_contexts:
-        print("Test gitlab configuration with context: ", context)
-        config = generate_gitlab_full_configuration(ctx, ".gitlab-ci.yml", dict(context))
+    has_errors = False
+
+    print(f'{color_message("info", Color.BLUE)}: Fetching Gitlab CI configurations...')
+    configs = get_all_gitlab_ci_configurations(ctx)
+
+    def test_gitlab_configuration(entry_point, init_config, context=None):
+        nonlocal has_errors
+
+        # Update config and lint it
+        config = generate_gitlab_full_configuration(ctx, entry_point, context=context, input_config=init_config)
```

Comment:
Nit: should we just call it `input_config` everywhere? It seems weird for the same variable to have two different names depending on the method.